File: Crashkurs Automaten/1 Einführung in Automaten - DFA/1 Getraenkeautomat1und2EuroDFA CS/Getraenkeautomat1und2EuroDFA/BottleMachine.cs
namespace Getraenkeautomat1und2EuroDFA;


// repräsentiert den Mineralwasserflaschengeräteautomat
public class BottleMachine
{
    public event EventHandler<Input> OnInput;

    public void Run()
    {
        while (true)    
        {
            Console.Write("Welche Münze wird eingeworfen? 1 oder 2?");

            string text = Console.ReadLine();

            Input input = text switch
            {
                "1" => Input._1EURO,
                "2" => Input._2EURO,
            };
            
            OnInput?.Invoke(this, input);
        }
    }
}

File: Crashkurs Automaten/1 Einführung in Automaten - DFA/1 Getraenkeautomt1und2EuroDFA Java/Getraenkeautomat1und2EuroDFA/src/main/java/at/htldonaustadt/dfa/devices/BottleMachine.java
package at.htldonaustadt.dfa.devices;

import at.htldonaustadt.dfa.finitestatemachine.*;

import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class BottleMachine {

    private Set<Event> subscribers = new HashSet<>();

    public void subscribe(Event subscriber) {
        if (subscriber != null)
            subscribers.add(subscriber);
    }

    public void unsubscribe(Event subscriber) {
        subscribers.remove(subscriber);
    }

    public void Run() {
        try (Scanner scanner = new Scanner(System.in)) {

            while (true) {
                System.out.print("Welche Münze wird eingeworfen? 1 oder 2? ");

                Input input = switch (scanner.next()) {
                    case "1" -> Input.EURO1;
                    case "2" -> Input.EURO2;
                    default -> throw new IllegalStateException("Invalid input!");
                };

                onInput(input);
            }
        }
    }

    public void onInput(Input input) {
        for (Event subscriber : subscribers) {
            subscriber.processInput(input);
        }
    }
}
